src/Development/IDE/Core/Rules.hs

```diff
@@ -457,21 +462,15 @@ getHieFileRule = define $ \GetHieFile f -> do
 
 getHiFileRule :: Rules ()
 getHiFileRule = defineEarlyCutoff $ \GetHiFile f -> do
-  session <- hscEnv <$> use_ GhcSession f
   logger  <- actionLogger
   -- get all dependencies interface files, to check for freshness
-  (deps,_)<- use_ GetLocatedImports f
-  depHis  <- traverse (use GetHiFile) (mapMaybe (fmap modLocationToNormalizedFilePath . snd) deps)
-
-  -- TODO find the hi file without relying on the parsed module
-  --      it should be possible to construct a ModSummary parsing just the imports
-  --      (see HeaderInfo in the GHC package)
-  pm      <- use_ GetParsedModule f
+  (deps,_) <- use_ GetLocatedImports f
```

Comment:
`GetLocatedImports` depends on `GetParsedModule` too, so in order to achieve the intended savings, it needs to be rewritten to use `HeaderInfo.getImports`